tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx

```diff
@@ -279,34 +280,11 @@ public:
    /// Throws an exception if the field is null.
    void AddField(std::unique_ptr<Detail::RFieldBase> field);
 
-   /// Throws an exception if fromWhere is null.
-   template <typename T>
-   void AddField(const NameWithDescription_t &fieldNameDesc, T* fromWhere) {
-      EnsureNotFrozen();
-      EnsureNotBare();
-      if (!fromWhere)
-         throw RException(R__FAIL("null field fromWhere"));
-      EnsureValidFieldName(fieldNameDesc.fName);
-
-      auto field = std::make_unique<RField<T>>(fieldNameDesc.fName);
-      field->SetDescription(fieldNameDesc.fDescription);
-      fDefaultEntry->AddValue(field->BindValue(std::shared_ptr<void>(fromWhere, [](void *) {})));
-      fFieldZero->Attach(std::move(field));
-   }
-
    /// Adds a top-level field based on existing fields. The mapping function is called with the qualified field names
    /// of the provided field and the subfields.  It should return the qualified field names used as a mapping source.
    /// Projected fields can only be used for models used to write data.
    RResult<void> AddProjectedField(std::unique_ptr<Detail::RFieldBase> field,
                                    std::function<std::string(const std::string &)> mapping);
-
-   template <typename T>
-   T *Get(std::string_view fieldName) const
```

Comment:
Would it be an idea to have a `Get<T>` (perhaps named differently) that returns a shared pointer to the object? The use case would be for type-erased fields. Right now the (only) way to do it is through `ntuple->GetModel().GetDefaultEntry()->GetPtr<T>("fieldname")`, which feels a bit unwieldy.

Reply:
That's a possibility, a getter acting as a shortcut for `GetModel().GetDefaultEntry()->GetPtr<T>`. On the other hand, callers can could also first get the entry and then call (multiple times) `entry.GetPtr<T>()`. How about we see if users stumble across it and defer the addition of the method until then?

Reply:
Yes fair enough, sounds good!